### PR TITLE
Generic Analytics Adapter: initial release

### PR DESCRIPTION
**Pull request by dgirardi**
## Type of change

- [x] Feature

## Description of change

This is a new generic analytics adapter, meant for publishers that have their own infrastrucutre for collecting analytics data. The adapter is "vendorless" for the purposes of GDPR enforcement (in the same way that sharedId is).

## Other information

Closes https://github.com/prebid/Prebid.js/issues/8835
Closes https://github.com/prebid/Prebid.js/issues/6564
Documentation: https://github.com/prebid/prebid.github.io/pull/4092

### CODE REVIEW

File: modules/genericAnalyticsAdapter.js [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+    {
+      enableAnalytics(config) {
+        if (optionsAreValid(config?.options || {})) {
+          options = Object.assign({}, DEFAULTS, config.options);
+          handler = options.handler || defaultHandler(options);
+          translate = translator(options.events);
+          parent.enableAnalytics.call(this, config);
+        }
+      },
+      track(event) {
```
Comment (patmmccann) [inline modules/genericAnalyticsAdapter.js:114]: is it straightforward to see which events a particular analytics adapter subscribes to?

File: modules/genericAnalyticsAdapter.js [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+
+  function processBatch() {
+    const currentBatch = batch;
+    batch = [];
+    callDepth++;
+    try {
+      // the pub-provided handler may inadvertently cause an infinite chain of events;
+      // even just logging an exception from it may cause an AUCTION_DEBUG event, that
+      // gets back to the handler, that throws another exception etc.
+      // to avoid the issue, put a cap on recursion
```
Comment (patmmccann) [inline modules/genericAnalyticsAdapter.js:71]: these infinite loops have occurred before, eg https://github.com/prebid/Prebid.js/pull/7805/files ; should we more generally defend against them?

Comment (dgirardi) [reply modules/genericAnalyticsAdapter.js:71]: There is a more general prevention mechanism in https://github.com/prebid/Prebid.js/pull/9113. That also allows the pub to set which events should be forwarded to any analytics adapter. However, adapters can choose to ignore some or circumvent the default event tracking system; so in general, the only way to find out which events are tracked is reading code. For this particular adapter though control is entirely on the publisher.

**Comment by dgirardi** [comment]
Agreed offline to make this accept a gvlid to make it truly generic; we can keep the "generic" name until a better one materializes.

**Comment by dgirardi** [comment]
Pausing this with the intention of tackling https://github.com/prebid/Prebid.js/issues/6564 as the solution for making this "vendorless".

**Review by Fawke — APPROVED** [review]
LGTM! Good work.

**Pull request merged**